src/main/java/com/example/springfirsthw/service/impl/BookServiceImpl.java

```diff
@@ -1,27 +1,51 @@
 package com.example.springfirsthw.service.impl;
 
 import com.example.springfirsthw.domain.Book;
+import com.example.springfirsthw.dto.BookDto;
+import com.example.springfirsthw.dto.CreateBookRequestDto;
+import com.example.springfirsthw.exception.EntityNotFoundException;
+import com.example.springfirsthw.mapper.BookMapper;
 import com.example.springfirsthw.repositories.BookRepository;
 import com.example.springfirsthw.service.BookService;
 import java.util.List;
-import org.springframework.beans.factory.annotation.Autowired;
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 
+@RequiredArgsConstructor
 @Service
 public class BookServiceImpl implements BookService {
+    private static final String NO_BOOK_BY_ID_FOUND_MSG = "Can't find book by id: ";
+
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+
+    public BookDto save(CreateBookRequestDto bookDto) {
+        return bookMapper
+                .toBookDto(bookRepository.save(bookMapper.toModel(bookDto)));
+    }
 
-    @Autowired
-    BookServiceImpl(BookRepository bookRepository) {
-        this.bookRepository = bookRepository;
+    public List<BookDto> getAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toBookDto)
+                .toList();
     }
 
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public BookDto getById(Long id) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(NO_BOOK_BY_ID_FOUND_MSG + id));
+        return bookMapper.toBookDto(book);
     }
 
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto requestDto) {
+        getById(id);
```

Comment:
Let's rewrite it with if statement.
```
if (bookRepository.findById(id).isEmpty()) {
  throw ...
}
```
I know that it is the same check as in `getById` method, but now it looks not obvious.